### PR TITLE
Feature#2904

### PR DESCRIPTION
**Pull request by sunbufu**
Fixes [#2904](https://github.com/apache/incubator-shardingsphere/issues/2904).

Changes proposed in this pull request:

- add ShowFullSegment for ShowTablesStatement's full.
- config ShowLikeSegment for ShowTablesStatement's pattern.
- config FromSchemaSegment for ShowTablesStatement's schema.
- add ShowTablesBackendHandler to execute and get query data.
- add some UT classes.

### CODE REVIEW

File: sharding-core/sharding-core-merge/src/main/java/org/apache/shardingsphere/core/merge/dal/show/ShowTablesMergedResult.java [hunk 1]
```diff
@@ -17,42 +17,43 @@
...
  * @author zhangliang
  * @author panjuan
+ * @author sunbufu
  */
-public final class ShowTablesMergedResult extends LogicTablesMergedResult {
+@RequiredArgsConstructor
+public final class ShowTablesMergedResult extends LocalMergedResultAdapter {
     
-    private static final Map<String, Integer> LABEL_AND_INDEX_MAP = new HashMap<>(1, 1);
+    private final List<String> tableNames;
```
Comment (terrymanu) [inline sharding-core/sharding-core-merge/src/main/java/org/apache/shardingsphere/core/merge/dal/show/ShowTablesMergedResult.java]: What is the different of tableNames & tableTypes?
Should they split to two class to handler different SQL?

File: sharding-core/sharding-core-parse/sharding-core-parse-mysql/src/main/antlr4/imports/mysql/DALStatement.g4 [hunk 2]
```diff
@@ -73,4 +73,8 @@ showWhereClause_
 
 setVariable
     : SET variable_?
+    ;
+
+showFull
+    : FULL?
     ;
```
Comment (terrymanu) [inline sharding-core/sharding-core-parse/sharding-core-parse-mysql/src/main/antlr4/imports/mysql/DALStatement.g4]: Need add blank line here

Comment (sunbufu) [reply sharding-core/sharding-core-parse/sharding-core-parse-mysql/src/main/antlr4/imports/mysql/DALStatement.g4]: ok

**Comment by sunbufu** [comment]
We can't implement this feature in sharding-jdbc for now.

**Pull request closed without merging**